src/CachePeer.cc

```diff
@@ -114,6 +117,208 @@ CachePeer::connectTimeout() const
+        os << " connect-timeout=" << connect_timeout_raw;
+
+    if (connect_fail_limit != PEER_TCP_MAGIC_COUNT)
+        os << " connect-fail-limit=" << connect_fail_limit;
+
+#if USE_CACHE_DIGESTS
+    if (digest_url)
+        os << " digest-url=" << digest_url;
+#endif
+
+    if (options.allow_miss)
+        os << " allow-miss";
+
+    if (options.no_tproxy)
+        os << " no-tproxy";
+
+    if (max_conn > 0)
+        os << " max-conn=" << max_conn;
+
+    if (standby.limit > 0)
+        os << " standby=" << standby.limit;
+
+    if (options.originserver)
+        os << " originserver";
+
+    if (domain)
+        os << " forceddomain=" << domain;
+
+    if (connection_auth == 0)
+        os << " connection-auth=off";
+    else if (connection_auth == 1)
+        os << " connection-auth=on";
+    else if (connection_auth == 2)
+        os << " connection-auth=auto";
+
+    secure.dumpCfg(os, "tls-");
+}
+
+void
+CachePeer::reportStatistics (std::ostream& os)
+{
+    os <<
+       "- " << name << ":\n" <<
+       "  type: " << typeString() << '\n' <<
+       "  host: " << host << '/' << http_port << '/' << icp.port << '\n';
+    os << "  flags:";
+    dumpOptions(os);
+    os << '\n';
+
+    std::vector<Ip::Address> addr;
+    std::copy(addresses, addresses+n_addresses, std::back_inserter(addr));
+    os << "  addresses: [ " << AsList(addr).prefixedBy("\"").delimitedBy("\", \"").suffixedBy("\"") << " ]\n";
+
+    os << "  status: " << (neighborUp(this) ? "Up" : "Down") << '\n' <<
+       "  fetches: " << stats.fetches << '\n' <<
+       "  open connections: " << stats.conn_open << '\n' <<
+       "  average RTT: " << stats.rtt << " msec\n";
+
+    if (stats.last_query > 0) {
+        os << "  last query:" <<
+           (squid_curtime - stats.last_query) << " seconds ago\n";
+    }
+
+    if (stats.last_reply > 0) {
+        os << " last reply: " <<
+           (squid_curtime - stats.last_reply) << " seconds ago\n";
+    }
+
+    os << "  pings sent: "  << stats.pings_sent << '\n' <<
+       "  pings acked: " << stats.pings_acked << " " <<
+       Math::intPercent(stats.pings_acked, stats.pings_sent) << "%\n";
+
+    os << "  replies ignored: " << stats.ignored_replies << " " <<
+       Math::intPercent(stats.ignored_replies, stats.pings_acked) << "%\n";
+
+    auto sectionHeader = AtMostOnce("  histogram of pings acked:\n");
+
+#if USE_HTCP
+    if (options.htcp) {
+        os << sectionHeader;
+        os << "    htcp misses: " << htcp.counts[0] << " " <<
+           Math::intPercent(htcp.counts[0], stats.pings_acked) << "%\n" <<
+           "    htcp hits: " << htcp.counts[1] << " " <<
+           Math::intPercent(htcp.counts[1], stats.pings_acked) << "%\n";
+    } else {
+#endif
+        for (auto op : WholeEnum<icp_opcode>()) {
+            if (icp.counts[op] == 0)
+                continue;
+
+            os << sectionHeader;
+            os << "    " <<
+               std::right << icp_opcode_str[op] << ": " <<
```

Comment:
Specifying right/left alignment only makes sense when also setting output field width, right?

```suggestion
            os << "    " << icp_opcode_str[op] << ": " <<
```